pkg/platform/automate/share.go

```diff
@@ -0,0 +1,390 @@
+	var result *corev1.ConfigMap
+	configmaps, err := client.CoreV1().ConfigMaps(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return result, err
+	}
+
+	for _, configMap := range configmaps.Items {
+		labels := configMap.GetLabels()
+		annotations := configMap.GetAnnotations()
+
+		if annotations["dolittle.io/microservice-id"] != microserviceID {
+			continue
+		}
+
+		if labels["environment"] != environment {
+			continue
+		}
+
+		if !strings.HasSuffix(configMap.GetName(), "-dolittle") {
+			continue
+		}
+
+		return &configMap, nil
+	}
+	return result, errors.New("not.found")
+}
+
+func GetDeployments(ctx context.Context, client kubernetes.Interface, namespace string) ([]appsv1.Deployment, error) {
+	deployments, err := client.AppsV1().Deployments(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var microserviceDeployments []appsv1.Deployment
+	for _, deployment := range deployments.Items {
+		if _, ok := deployment.Labels["microservice"]; !ok {
+			continue
+		}
+		if _, ok := deployment.Annotations["dolittle.io/microservice-id"]; !ok {
+			continue
+		}
+		microserviceDeployments = append(microserviceDeployments, deployment)
+	}
+	return microserviceDeployments, nil
+}
+
+// GetRuntimeDeployment get's a microservices deployment and the index of the Runtime container within that deployment
+func GetRuntimeDeployment(ctx context.Context, client kubernetes.Interface, applicationID string, environment string, microserviceID string) (int, appsv1.Deployment, error) {
+	namespace := fmt.Sprintf("application-%s", applicationID)
+
+	deployments, err := GetDeployments(ctx, client, namespace)
+	if err != nil {
+		return -1, appsv1.Deployment{}, err
+	}
+
+	for _, deployment := range deployments {
+		if deployment.Annotations["dolittle.io/microservice-id"] != microserviceID {
+			continue
+		}
+
+		if deployment.Labels["environment"] != environment {
+			continue
+		}
+
+		for index, container := range deployment.Spec.Template.Spec.Containers {
+			if container.Name == "runtime" {
+				return index, deployment, nil
+			}
+		}
+	}
+
+	return -1, appsv1.Deployment{}, errors.New("not.found")
+}
+
+func ConvertObjectMetaToMicroservice(objectMeta metav1.Object) configK8s.Microservice {
+	labels := objectMeta.GetLabels()
+	annotations := objectMeta.GetAnnotations()
+
+	microserviceID := annotations["dolittle.io/microservice-id"]
+	microserviceName := labels["microservice"]
+	customerTenant := configK8s.Tenant{
+		Name: labels["tenant"],
+		ID:   annotations["dolittle.io/tenant-id"],
+	}
+	k8sApplication := configK8s.Application{
+		Name: labels["application"],
+		ID:   annotations["dolittle.io/application-id"],
+	}
+
+	environment := labels["environment"]
+
+	var kind platform.MicroserviceKind
+	if kindString, ok := annotations["dolittle.io/microservice-kind"]; !ok {
+		kind = platform.MicroserviceKindUnknown
+	} else {
+		kind = platform.MicroserviceKind(kindString)
+	}
+
```

Comment:
The use of else is not needed here. I guess it might be a style thing, but we have a default "unknown" and we plan on changing it "if" something is different.

We also have very similar logic in https://github.com/dolittle/platform-api/blob/c978b7ba019986fe33a3077064facb14dc6c1e31/pkg/platform/k8s_repo.go#L199-L202

Which makes me believe we should hoist this into a func, and then clean up the code in both places.
To start with, I think the signature could be "annotations". and return kind, if its finds it, or unknown.


```suggestion
	kind := platform.MicroserviceKindUnknown
	if kindString, ok := annotations["dolittle.io/microservice-kind"]; ok {
		kind = platform.MicroserviceKind(kindString)
	}```

Reply:
Yeah I was debating myself that should I be setting the variable first by default to unknown or using an if/else.

I created function `GetMicroserviceKindFromAnnotations` in the platform package. I couldn't make it inside `automate` as we'd get a cyclical import (`automate` imported the `platform.MicroserviceKind` type, `platform` imported `automate` to get the kind -> cyclical import).